persistence/javadsl/src/main/scala/com/lightbend/lagom/internal/javadsl/persistence/ReadSideImpl.scala

```diff
@@ -97,23 +97,30 @@ private[lagom] class ReadSideImpl @Inject()(
           config.randomBackoffFactor
         )
 
-      val readSideProps =
+      val streamName    = tags.head.eventType.getName
+      val projectorName = readSideName
+
+      val readSidePropsFactory = (projectorRegistryActorRef: ActorRef) =>
+        // TODO: use the actorRef on the ReadSideActor to register, ping-back info, etc...
         ReadSideActor.props(
+          streamName,
+          projectorName,
           config,
           eventClass,
           globalPrepareTask,
-          registry.eventStream[Event],
-          processorFactory
+          persistentEntityRegistry.eventStream[Event],
+          processorFactory,
+          projectorRegistryActorRef
         )
 
-      val shardingSettings = ClusterShardingSettings(system).withRole(config.role)
-
-      ClusterDistribution(system).start(
+      projectorRegistryImpl.registerProjectorGroup(
+        tags.head.eventType.getName, // TODO: use the name from the entity, not the tags
```

Comment:
After a closed look, why we need a `streamName` passed? 

The readSideName must be unique and should suffice.

Reply:
I thought it'd be useful to know where a projector worker is connected to. After all the projector does have that information inside the `tagname` so why not make it a little more _accessible_?